### ISSUE TITLE
Dynamic batching?

### ISSUE THREAD

**Issue by keko950** [AlexeyAB/darknet#3924]
Is there any way to set dynamic batching at inference time? Since set_batch_network is a private function is there any way to do it? I could modify the source code myself, but i really dont wan't to, since i want to stay updated with this repo.

**Comment by keko950** [comment]
I updated my pull request in order to add set_batch_network to LIB_API https://github.com/AlexeyAB/darknet/pull/3465
Would be useful i think.. what do you think? @AlexeyAB

**Comment by AlexeyAB** [comment]
set_batch_network() only setups cuDNN descriptors, but doesn't realocate GPU memory, so it can be used only to reduce batch-size.

this problem comes from the original repo: https://github.com/pjreddie/darknet/blob/61c9d02ec461e30d55762ec7669d6a1d3c356fb2/src/network.c#L339-L356

So it's better to free_network() and create new network.

**Comment by keko950** [comment]
Hmm.. but that is a time consuming method. 

Btw, im trying also to use a fixed size batch but cant manage to make it work. I can only access the data related to the first image predicted in the batch, its more like the output layer doesnt increase correctly.. Im doing this just in the end of the batch for loop

_network->layers[_network->n-1].output = _network->layers[_network->n-1].output + 512 * 512* 3;

But when i call get_network_boxes it returns the same boxes each time..

**Comment by keko950** [comment]
Yeah, definitely its something going wrong with layers.. just printed  
        PRINT(l.h << "h")
        PRINT(l.w << "w")
        PRINT(l.n << "n")
        PRINT(l.classes << "classes")
        PRINT(l.coords << "coords")
of last layer (yolo), and all this elements have value 0..

**Comment by keko950** [comment]
So.. apparently just before return the net in function load_network_custom() network.layers[71].type = 26 but if i call network.layers[71].type in my code after return this is equal to 0.. hm..
